### ISSUE TITLE
Document when Windows.System.LauncherOptions.TargetApplicationPackageFamilyName property is required to be set

### ISSUE THREAD

**Issue by metathinker** [MicrosoftDocs/winrt-api#994]
The documentation page for the above-named [TargetApplicationPackageFamilyName property of the Windows.System.LauncherOptions data record](https://docs.microsoft.com/uwp/api/windows.system.launcheroptions.targetapplicationpackagefamilyname) states that it is optional. This is not always true.

That is, in some cases, when you call Windows.System.Launcher static methods to launch an app associated with a file type or URI protocol, you _must_ specify exactly which app you want to be launched by passing a LauncherOptions record with this property set to the package family name for the app package containing that app.

2 such cases are as follows:

1. You want to launch an app with a URI and pass extra input data to the launched app, using [Launcher.LaunchUriAsync(Uri, LauncherOptions, ValueSet)](https://docs.microsoft.com/uwp/api/windows.system.launcher.launchuriasync#Windows_System_Launcher_LaunchUriAsync_Windows_Foundation_Uri_Windows_System_LauncherOptions_Windows_Foundation_Collections_ValueSet_).
1. You want to launch an app with a URI and wait for the app to send output data back to your app, using any overload of [Launcher.LaunchUriForResultsAsync()](https://docs.microsoft.com/uwp/api/windows.system.launcher.launchuriforresultsasync).

In those cases, both the calling app and launched app need to agree on how to interpret the extra data passed from one app to the other, so it may not make sense for the calling app to launch whatever app happens to be the default for that URI protocol. I assume that's why Windows doesn't let calling apps do that.

Please update this documentation page to say when this property is required to be set, and update other reference and guide doc pages to provide similar information where it may be useful.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2ff12cee-d588-9e46-2d5e-844dbe6f711a
* Version Independent ID: 2f14a1b3-c1f5-6dbd-4675-aceff296cfa0
* Content: [LauncherOptions.TargetApplicationPackageFamilyName Property (Windows.System) - Windows UWP applications](https://docs.microsoft.com/en-us/uwp/api/windows.system.launcheroptions.targetapplicationpackagefamilyname)
* Content Source: [https://cpubwin.visualstudio.com/DefaultCollection/windows-uwp/_git/winrt-api-build?path=%2fwinrt-api-build%2fwindows.system.launcheroptions.yml&version=GBlive&_a=contents](https://cpubwin.visualstudio.com/DefaultCollection/windows-uwp/_git/winrt-api-build?path=%2fwinrt-api-build%2fwindows.system.launcheroptions.yml&version=GBlive&_a=contents)
* Product: **uwp**
* Git Login: **hickeys**
* Microsoft Alias: **hickeys**

**Comment by metathinker** [comment]
It also might make sense to better distinguish this property from the similarly-named [**Preferred**ApplicationPackageFamilyName](https://docs.microsoft.com/en-us/uwp/api/windows.system.launcheroptions.preferredapplicationpackagefamilyname) and [**Preferred**ApplicationDisplayName](https://docs.microsoft.com/en-us/uwp/api/windows.system.launcheroptions.preferredapplicationdisplayname) properties.

Those properties can be set when you _don't_ require one exact app to open a specific file or URI, but you would like to recommend an app in the "How do you want to open this file?" dialog that Windows for desktop shows when no suitable app is installed on the device.

Right now, the distinction is only implicit in the documentation itself; it might make sense to make it more explicit - such as by making the doc pages for PreferredApplicationPackageFamilyName and TargetApplicationPackageFamilyName more similar, while calling out the differences.

**Comment by jbrianceau** [comment]
For the record, I've just filed a new issue for this (#1696), I didn't see that this doc problem was already reported here.

**Comment by alvinashcraft** [comment]
Thanks for submitting your feedback and we apologize for the delayed response. I see that updates have not been made since the creation of this Issue, so I went ahead and created an internal PR to incorporate your suggestions. The changes have been merged and should be live soon, so I will go ahead and close this and circle back when the changes are live. 

Though the Issue will be marked as Closed, we will monitor any replies that come afterwards so treat this like an open Issue. Many thanks!
